src/proj04BittingEllmerWang/Controller.java

```diff
@@ -110,24 +142,45 @@ private void handleOpenMenuItem(Event event) throws IOException {
         );
         File selectedFile = fileChooser.showOpenDialog(tabPane.getScene().getWindow());
 
+
         // if a valid file is selected
         if (selectedFile != null) {
             // get the path of the file selected
             String filePath = selectedFile.getPath();
+
+            // check if the file has already been opened in tabPane.
+            for (Object tabObj : tabPane.getTabs().toArray()) {
+                Tab tab = (Tab) tabObj;
+                if (tab.getId().equals(filePath)){
+                    // if so, switch to existing tab.
+                    SingleSelectionModel<Tab> selectionModel = tabPane.getSelectionModel();
+                    selectionModel.select(tab); // select by object
+                    return;
+                }
+            }
+
             // read the content of the file to a string
             String fileContent = new String(Files.readAllBytes(Paths.get(filePath)));
             // generate a new tab and put the file content into the text area
             handleNewMenuItem(event); // TODO: probably rename the method as this isn't handling this event
             // get the current textBox
             Tab currentTab = tabPane.getSelectionModel().getSelectedItem();
-            TextArea textBox = (TextArea) currentTab.getContent();
-            // set the content of the textBox
-            textBox.setText(fileContent);
+            CodeArea codeBox = (CodeArea)((VirtualizedScrollPane) currentTab.getContent()).getContent();
+            // set the content of the codeBox
+            codeBox.appendText(fileContent);
             // set the title of the tab
             String[] fileAncestors = filePath.split("/");
             currentTab.setText(fileAncestors[fileAncestors.length - 1]);
             currentTab.setId(filePath);
         }
+
+        // re-enable the buttons when there are tabs
+        if (!tabPane.getTabs().isEmpty()) {
```

Comment:
This code block is re-used, so there should definitely be an enableAll method. That method should be called if the method actually opens a new tab.